### ISSUE TITLE
mouse can't show

### ISSUE THREAD

**Issue by Mr-LLLLL** [neovide/neovide#1966]
**Describe the bug**
enter neovide, move the mouse, cant see any mouse cursor, and click mouse, it can work, just not see any mouse cursor

**Comment by fredizzimo** [comment]
Neovide is using the default operating system mouse cursor, so it should show. 

But there might be a bug in the hiding behaviour, do you have `neovide_hide_mouse_when_typing` enabled? 
 https://neovide.dev/configuration.html#hiding-the-mouse-when-typing

**Comment by Mr-LLLLL** [comment]
yes, i set neovide_hide_mouse_when_typing is true, maybe my talk is not clear, the bug i meet is the cursor can't show when i move the cursor with mouse

**Comment by Mr-LLLLL** [comment]
when i set 

> Could you try with `neovide_hide_mouse_when_typing=false` and see if the problem goes away. We need to figure out if the problem is with that or something else.

NO.

when i with `neovide_hide_mouse_when_typing=false`, open the neovide, the cursor not show anywhere, and move it by mouse, then not show.

**Comment by Mr-LLLLL** [comment]
i think the bug is mouse can't show in screen, but cursor's click function is ok,  whaterver `neovide_hide_mouse_when_typing` value.
and thank you @fredizzimo. 
my english is bad, and i just start studing rust, have not capable to fix it.

**Comment by Mr-LLLLL** [comment]
no.
with `neovide_hide_mouse_when_typing=false`:
if cursor show before open neovide, the cursor is yet show, not hide anymore, no matter how to operate in neovide.
if cursor hide before open neovide, the cursor is not show yet, not show anymore, no matter how to operate in neovide.

with `neovide_hide_mouse_when_typing=ture`:
if cursor show before open neovide, hit any key, the cursor will hide,then whatervery you do in neovide(include move th mouse), the cursor not show again. 
if cursor hide before open neovide, the cursor will be hide,  no matter how to operate in neovide.

**Comment by fredizzimo** [comment]
Ok, I understand now. But two questions?
1. How do you hide the cursor before starting Neovide?
2. Are you on macOS? There's this winit bug that could explain it https://github.com/rust-windowing/winit/issues/2539. And it seems to only happen on macOS 13. Older versions are not affected. It could also be an operating system bug, that needs a workaround in Winit.

**Comment by Mr-LLLLL** [comment]
1.I open neovide is in terminal, terminal always hide cursor
2.i use ubuntu 22.4

**Comment by fredizzimo** [comment]
Ah, ok. I will try to repeat it myself a bit later.

It looks like the bug might be here https://github.com/neovide/neovide/blob/main/src/window/mouse_manager.rs#L442. We use our own state for tracking if the cursor should be hidden or not. But maybe the state is global for the whole desktop environment, in that case we need to ensure that we always show the cursor when both going in and out from the window.

**Comment by fredizzimo** [comment]
This seems related, and is probable caused by the same issue

* https://github.com/neovide/neovide/issues/1620

**Comment by fredizzimo** [comment]
#2414 has more information
